lib/rubocop/cop/rspec/repeated_include_example.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for repeated include of shared examples.
+      #
+      # @example
+      #
+      #    # bad
+      #    describe 'foo' do
+      #      include_examples 'cool stuff'
+      #      include_examples 'cool stuff'
+      #    end
+      #
+      #    # bad
+      #    describe 'foo' do
+      #      it_behaves_like 'a cool', 'thing'
+      #      it_behaves_like 'a cool', 'thing'
+      #    end
+      #
+      #    # bad
+      #    context 'foo' do
+      #      it_should_behave_like 'a duck'
+      #      it_should_behave_like 'a duck'
+      #    end
+      #
+      #    # good
+      #    describe 'foo' do
+      #      include_examples 'cool stuff'
+      #    end
+      #
+      #    describe 'bar' do
+      #      include_examples 'cool stuff'
+      #    end
+      #
+      #    # good
+      #    describe 'foo' do
+      #      it_behaves_like 'a cool', 'thing'
+      #      it_bahaves_like 'a cool', 'person'
+      #    end
+      #
+      #    # good
+      #    context 'foo' do
+      #      it_should_behave_like 'a duck'
+      #      it_should_behave_like 'a goose'
+      #    end
+      #
+      #    # good
+      #    context 'foo' do
+      #      cool = 'neat'
+      #      include_examples 'something', cool
+      #      cool = 'fun'
+      #      include_examples 'something', cool
+      #    end
+      #
+      class RepeatedIncludeExample < Base
+        MSG = 'Repeated include of shared_examples %<name>s on line(s) %<loc>s'
+
+        def_node_matcher :several_include_examples?, <<-PATTERN
+          (begin <#include_examples? #include_examples? ...>)
+        PATTERN
+
+        def_node_matcher :include_examples?,
+                         Includes::EXAMPLES.send_or_block_or_block_pass_pattern
```

Comment:
Speaking of blocks and block-passes, what is the behavior for:
```ruby
it_behaves_like 'foo' do
  let(:bar) { 1 }
end
it_behaves_like 'foo' do
  let(:bar) { 2 }
end
```
and
```ruby
it_behaves_like 'foo', &bar
it_behaves_like 'foo', &bar
# those two seem like duplicates, but imagine it has side effects
bar = -> { $incr += 1 }
```

Reply:
Will just go for literals / constants now, so a `block_pass` will be allowed without offense.

Reply:
Also removed the matching of blocks as I can't see a need to detect ones that look similar to a static analysis.